### PR TITLE
RATIS-1517. Remove core Ratis protos

### PR DESCRIPTION
**Pull request by adoroszlai**
## What changes were proposed in this pull request?

 * Move `Hadoop.proto` to the `ratis-hadoop` submodule and update its POM to compile it.
 * Remove core Ratis protos and `ratis-proto` submodule from this repo.
 * Add missing Mockito dependency, found by trying to run the tests (currently disabled).

https://issues.apache.org/jira/browse/RATIS-1517

## How was this patch tested?

Temporarily enabled tests to verify compiled classes are usable.

CI:
https://github.com/adoroszlai/ratis-hadoop-projects/actions/runs/1780584551

### CODE REVIEW

File: ratis-hadoop/pom.xml [hunk 1]
```diff
@@ -100,24 +110,98 @@
...
+              <protocArtifact>
+                com.google.protobuf:protoc:${shaded.protobuf.version}:exe:${os.detected.classifier}
+              </protocArtifact>
+              <includes>
+                <include>Hadoop.proto</include>
+              </includes>
+              <!-- With multiple executions, this must be `false` otherwise we wipe out the previous execution -->
+              <clearOutputDirectory>false</clearOutputDirectory>
+            </configuration>
+          </execution>
```
Comment (szetszwo) [inline ratis-hadoop/pom.xml]: >compile-grpc can be removed.

File: ratis-hadoop/pom.xml [hunk 2]
```diff
@@ -100,24 +110,98 @@
...
+            <phase>process-sources</phase>
+            <goals>
+              <goal>replace</goal>
+            </goals>
           </execution>
         </executions>
+        <configuration>
+          <basedir>${project.build.directory}/generated-sources</basedir>
+          <includes>
+            <include>**/grpc-java/**/*.java</include>
```
Comment (szetszwo) [inline ratis-hadoop/pom.xml]: We should remove this include for grpc-java.

File: ratis-hadoop/pom.xml [hunk 3]
```diff
@@ -100,24 +110,98 @@
...
+          </includes>
+          <replacements>
+            <replacement>
+              <token>([^\.])com.google</token>
+              <value>$1org.apache.ratis.thirdparty.com.google</value>
+            </replacement>
+            <replacement>
+              <token>([^\.])io.grpc</token>
+              <value>$1org.apache.ratis.thirdparty.io.grpc</value>
+            </replacement>
```
Comment (szetszwo) [inline ratis-hadoop/pom.xml]: We should also remove this grpc replacement.

**Review by szetszwo — COMMENTED** [review]
@adoroszlai , thanks a lot working on this.  There are a few grpc related items.  We should remove them.  The change looks good other than that.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by adoroszlai** [comment]
Thanks @szetszwo for reviewing and committing it.

**Pull request merged**